AzureMonitorAgent/ama_tst/modules/errors.py

```diff
@@ -0,0 +1,103 @@
+import copy
+import subprocess
+
+from error_codes import *
+
+# backwards compatible input() function for Python 2 vs 3
+try:
+    input = raw_input
+except NameError:
+    pass
+
+# error info edited when error occurs
+error_info = []
+
+# list of all errors called when script ran
+err_summary = []
+
+
+
+# # set of all errors which are actually warnings
+warnings = set([WARN_FILE_PERMS, WARN_LOG_ERRS, WARN_LOG_WARNS, WARN_LARGE_FILES, \
+                WARN_INTERNET_CONN, WARN_INTERNET, WARN_ENDPT])
+
+# dictionary correlating error codes to error messages
+error_messages = {
+#     ERR_SUDO_PERMS : "Couldn't access {0} due to inadequate permissions. Please run the troubleshooter "\
+#           "as root in order to allow access.",
+    ERR_FOUND : "Please go through the output above to find the errors caught by the troubleshooter.",
+    ERR_BITS : "Couldn't get if CPU is not 32-bit or 64-bit.",
+    ERR_OS_VER : "This version of {0} ({1}) is not supported. Please download {2}. To see all "\
+          "supported Operating Systems, please go to:\n"\
+          "\n   https://docs.microsoft.com/en-us/azure/azure-monitor/agents/agents-overview#linux\n",
+    ERR_OS : "{0} is not a supported Operating System. To see all supported Operating "\
+          "Systems, please go to:\n"\
+          "\n   https://docs.microsoft.com/en-us/azure/azure-monitor/agents/agents-overview#linux\n",
+    ERR_FINDING_OS : "Coudln't determine Operating System. To see all supported Operating "\
+          "Systems, please go to:\n"\
+          "\n   https://docs.microsoft.com/en-us/azure/azure-monitor/agents/agents-overview#linux\n",
+    ERR_FREE_SPACE : "There isn't enough space in directory {0} to install OMS - there needs to be at least 500MB free, "\
+          "but {0} has {1}MB free. Please free up some space and try installing again.",
+    ERR_PKG_MANAGER : "This system does not have a supported package manager. Please install 'dpkg' or 'rpm' "\
+          "and run this troubleshooter again.",
+    ERR_MULTIPLE_AMA : "There are more than one AMA installed, please remove the extra AMA packages.",
+    ERR_AMA_INSTALL : "AMA package isn't installed correctly.",
+    ERR_SUBMODULE_INSTALL : "Submodule(s) {0} not installed correctly.",
+    ERR_LOG_DAEMON : "No logging daemon found. Please install rsyslog or syslog-ng.",
+    ERR_SYSLOG_USER : "Syslog user is not created successfully.",
+    ERR_OLD_AMA_VER : "You are currently running AMA Version {0}. This troubleshooter only "\
+          "supports versions 1.9 and newer. Please upgrade to the newest version. You can find "\
```

Comment:
Have you confirmed that the TST works all the way back to AMA version 1.9 (I believe current version out is 1.21)? If you don't want to test back that far, you should put the oldest version that you tested the TST on

Reply:
TST adds installation checks from AMA 1.21.0 so I believe this should be the good version to put on